### PR TITLE
added field frm_configs to MerchantConnectorAccount

### PR DESCRIPTION
**Pull request by rishavkar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

We need to accept frm_configs from the dashboard. So needed to do development for accepting new fields for endpoint: `/account/:account_id/connectors`

### Additional Changes

- [x] This PR modifies the database schema : 
                    `ALTER TABLE "merchant_connector_account" ADD COLUMN frm_configs jsonb;`
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

It a new feature...we are going to integrate FRM services into hyperswitch, for which we required additional fields for create connector request. 
This PR will accept frm_configs field for create connector request, and accordingly persist in the db.

## How did you test it?

Following are the steps:
- Adding an extra field to the request body, to existing curl.
- Getting 200 and Expecting this same body in response.
  ```
"frm_configs" : {
      "frm_enabled_pms" : ["card"],
      "frm_enabled_pm_types" : ["credit"],
      "frm_enabled_gateways" : ["stripe"],
      "frm_action": "cancel_txn",
      "frm_preferred_flow_type" : "pre"
  }
```

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/api_models/src/admin.rs [hunk 1]
```diff
@@ -345,8 +345,18 @@ pub struct MerchantConnector {
     /// You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Metadata is useful for storing additional, structured information on an object.
     #[schema(value_type = Option<Object>,max_length = 255,example = json!({ "city": "NY", "unit": "245" }))]
     pub metadata: Option<pii::SecretSerdeValue>,
+    pub frm_configs : Option<FrmConfigs>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
```
Comment (dracarys18) [inline crates/api_models/src/admin.rs:351]: Do we need Deserialize here?

File: crates/api_models/src/enums.rs [hunk 2]
```diff
@@ -766,3 +766,45 @@ impl From<AttemptStatus> for IntentStatus {
...
 }
+
+
+#[derive(
+    Clone,
+    Debug,
+    Eq,
+    Hash,
+    PartialEq,
+    serde::Deserialize,
```
Comment (dracarys18) [inline crates/api_models/src/enums.rs:777]: Do we need these derives?

File: crates/api_models/src/enums.rs [hunk 3]
```diff
@@ -766,3 +766,45 @@ impl From<AttemptStatus> for IntentStatus {
...
+    ManualReview
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Eq,
+    Hash,
+    PartialEq,
+    serde::Deserialize,
```
Comment (dracarys18) [inline crates/api_models/src/enums.rs:798]: Same here

File: crates/router/src/core/admin.rs [hunk 4]
```diff
@@ -309,7 +309,8 @@ pub async fn create_payment_connector(
             field_name: "connector_account_details".to_string(),
             expected_format: "auth_type and api_key".to_string(),
         })?;
-
+    let frm_value: serde_json::Value = utils::Encode::<api_models::admin::FrmConfigs>::encode_to_value(&req.frm_configs).expect("Error encoding frm_configs to value");
```
Comment (dracarys18) [inline crates/router/src/core/admin.rs:312]: Don't use expect.

File: crates/router/src/core/admin.rs [hunk 5]
```diff
@@ -419,7 +421,7 @@ pub async fn update_payment_connector(
             })
             .collect::<Vec<serde_json::Value>>()
     });
-
+    let frm_value: serde_json::Value = utils::Encode::<api::PaymentMethodsEnabled>::encode_to_value(&req.frm_configs).expect("Error encoding frm_configs to value");
```
Comment (dracarys18) [inline crates/router/src/core/admin.rs:424]: Don't use expect. Expect panics

**Review by dracarys18 — CHANGES_REQUESTED** [review]
Also please format the code

File: crates/api_models/src/enums.rs (hunk 2 above)
Comment (rishavkar) [reply crates/api_models/src/enums.rs:777]: We can remove Eq,Hash,PartialEq,

File: crates/api_models/src/admin.rs [hunk 6]
```diff
@@ -345,8 +345,18 @@ pub struct MerchantConnector {
...
+#[serde(deny_unknown_fields)]
+pub struct FrmConfigs {
```
[the remark this answers is not in the document]
Comment (rishavkar) [reply crates/api_models/src/admin.rs:353]: done

File: crates/api_models/src/admin.rs [hunk 7]
```diff
@@ -345,8 +345,18 @@ pub struct MerchantConnector {
     /// You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Metadata is useful for storing additional, structured information on an object.
     #[schema(value_type = Option<Object>,max_length = 255,example = json!({ "city": "NY", "unit": "245" }))]
     pub metadata: Option<pii::SecretSerdeValue>,
+    pub frm_configs : Option<FrmConfigs>,
```
[the remark this answers is not in the document]
Comment (rishavkar) [reply crates/api_models/src/admin.rs:348]: done

File: crates/api_models/src/enums.rs (hunk 3 above)
Comment (rishavkar) [reply crates/api_models/src/enums.rs:798]: We can remove Eq,Hash,PartialEq

File: crates/router/src/core/admin.rs (hunk 4 above)
Comment (rishavkar) [reply crates/router/src/core/admin.rs:312]: done

**Review by jagan-jaya — APPROVED** [review]

**Review by dracarys18 — APPROVED** [review]

File: crates/api_models/src/admin.rs [hunk 8]
```diff
@@ -345,8 +345,20 @@ pub struct MerchantConnector {
     /// You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Metadata is useful for storing additional, structured information on an object.
     #[schema(value_type = Option<Object>,max_length = 255,example = json!({ "city": "NY", "unit": "245" }))]
     pub metadata: Option<pii::SecretSerdeValue>,
+    pub frm_configs: Option<FrmConfigs>,
+    //contains the frm configs for the merchant... it should be of this format:: "\"frm_configs\" : { \"frm_enabled_pms\" : [\"card\"], \"frm_enabled_pm_types\" : [\"credit\"], \"frm_enabled_gateways\" : [\"stripe\"], \"frm_action\": \"cancel_txn\", \"frm_preferred_flow_type\" : \"pre\" }"
```
Comment (jarnura) [inline crates/api_models/src/admin.rs:349]: Add as documentation comments, use schema to examples

Comment (jarnura) [inline crates/api_models/src/admin.rs:349]:
```suggestion
    /// contains the frm configs for the merchant... it should be of this format:: "\"frm_configs\" : { \"frm_enabled_pms\" : [\"card\"], \"frm_enabled_pm_types\" : [\"credit\"], \"frm_enabled_gateways\" : [\"stripe\"], \"frm_action\": \"cancel_txn\", \"frm_preferred_flow_type\" : \"pre\" }"
```

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**